### ISSUE TITLE
get the unique key for every device

### ISSUE THREAD

**Issue by flash89y** [smouldery/shock-collar-control#25]
I wrote a similiar arduino implementation back in 02/2018
(https://github.com/flash89y/Arduino/tree/master/libraries/DogCollar)

Thanks to your code i understand that there is a unique key and where it is.
Today I added instructions and a sketch to find it :)

That is a thing that other uses may need very much (at a blog someone - you know who :P - linked your site and posted code bits. He talked about trying out many collar devices until one worked - that could be simplified with knowing the key :)

Maybe you write an own implementation OR you link to my sketch to find it and "21"->0 / "12"->1 to convert to your bytes.

**Comment by smouldery** [comment]
yeah absolutely! sorry about the slow response, only just saw this - your code looks awesome!

I'll probably look at working off of your code and implementing it directly in mine but i've no idea how long that'll take - in the mean time i've linked to it in the repo readme :)

**Comment by huppfi** [comment]
Hey, 

The repo of @flash89y is gone is there an update to it? My collar is not responding to the Arduino Code and I have no idea why.

**Comment by ryanchesler** [comment]
similar issue, not sure exactly how to troubleshoot. Seems like everything is configured correctly but no response from the collar. I assume something to do with this key. Seems there is some way to pair multiple devices to the remote so I would assume there are unique keys we have to find somehow.

**Comment by smouldery** [comment]
@huppfi   it's possible it's changed (it's been a LONG time since I purchased mine) - i'd need more details about your problem. Do you have the ability to capture the output of the stock antenna and compare to the one in the docs to make sure it uses the same protocol still? You can use an SDR for this if you have one, or a 433mhz receiver and "oscilloscope" style code for Arduino. 

@ryanchesler likely, yes - as above, if you can capture the waveform we can be more certain it's "just" the key. RE: Pairing - if you don't know the key, and don't care about using the remote, you can just reset the collar and re-pair to the PC transmitter not the remote. you only need to extract the key if you want to keep using the remote as well as the PC.
